build/cmake/MongoSettings.cmake

```diff
@@ -0,0 +1,284 @@
+include_guard(DIRECTORY)
+
+#[[ Bool: Set to TRUE if the environment variable MONGODB_DEVELOPER is a true value ]]
+set(MONGODB_DEVELOPER FALSE)
+
+# Detect developer mode:
+set(_is_dev "$ENV{MONGODB_DEVELOPER}")
+if(_is_dev)
```

Comment:
In most cases, yes, but this is the best way to avoid the wacky world of `if` parsing. The most predictable behavior is to activate if's `<variable>` parsing, but the only way to do that is with a bare identifier naming a variable. As such, I define a variable with the value of the env var, and then name that variable.